packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.CustomGroupHeaders.Example.tsx

```diff
@@ -68,32 +75,43 @@ export class DetailsListCustomGroupHeadersExample extends React.Component<{}, {}
     );
   }
 
-  private _onRenderDetailsHeader(props: IDetailsHeaderProps) {
-    return <DetailsHeader {...props} ariaLabelForToggleAllGroupsButton={'Toggle selection'} />;
-  }
+  private _onRenderDetailsHeader: IDetailsListProps['onRenderDetailsHeader'] = props => {
+    if (props) {
+      return <DetailsHeader {...props} ariaLabelForToggleAllGroupsButton={'Toggle selection'} />;
+    }
+    return null;
+  };
 
-  private _onRenderGroupHeader = (props: IGroupDividerProps): JSX.Element => {
-    return (
-      <div className={classNames.headerAndFooter}>
-        <div className={classNames.headerTitle}>{`Custom header for ${props.group!.name}`}</div>
-        <div className={classNames.headerLinkSet}>
-          <Link className={classNames.headerLink} onClick={this._onToggleSelectGroup(props)}>
-            {props.isSelected ? 'Remove selection' : 'Select group'}
-          </Link>
-          <Link className={classNames.headerLink} onClick={this._onToggleCollapse(props)}>
-            {props.group!.isCollapsed ? 'Expand group' : 'Collapse group'}
-          </Link>
+  private _onRenderGroupHeader: IDetailsGroupRenderProps['onRenderHeader'] = props => {
```

Comment:
Is this the new usage of the strictFunctionTypes? The `onRenderHeader`

Reply:
yes, instead of defining the function type via  `(props: IGroupDividerProps): JSX.Element`, I'm refering directly to the `onRenderHeader` from `IDetailsGroupRenderProps`. This ensures that the types are identical, therefor not causing issues when using strictFunctionTypes.

It also exposed a bug in that `props` are actually optional, and null should be returned if props don't exist.